python/ray/experimental/workflow/workflow_access.py

```diff
@@ -214,33 +257,36 @@ def actor_ready(self, actor_id: str) -> ray.ObjectRef:
                              "it has failed before initialization.")
         return self._actor_initialized[actor_id]
 
-    def get_output(self, workflow_id: str) -> ray.ObjectRef:
+    def get_output(self, workflow_id: str) -> "ray.ObjectRef":
         """Get the output of a running workflow.
 
         Args:
-            workflow_id: The ID of a  workflow job.
+            workflow_id: The ID of a workflow job.
 
         Returns:
             An object reference that can be used to retrieve the
             workflow result.
         """
         if workflow_id in self._workflow_outputs:
-            return self._workflow_outputs[workflow_id]
+            return self._workflow_outputs[workflow_id].output
         wf_store = workflow_storage.WorkflowStorage(workflow_id, self._store)
         meta = wf_store.load_workflow_meta()
         if meta is None:
             raise ValueError(f"No such workflow {workflow_id}")
         if meta == common.WorkflowStatus.FAILED:
             raise ValueError(
                 f"Workflow {workflow_id} failed, please resume it")
-        output = recovery.resume_workflow_job.remote(workflow_id,
-                                                     self._store.storage_url)
-        self._workflow_outputs[workflow_id] = output
+        step_id = wf_store.get_entrypoint_step_id()
+        result = recovery.resume_workflow_step(workflow_id, step_id,
+                                               self._store.storage_url)
+        latest_output = LatestWorkflowOutput(result.state, workflow_id,
+                                             step_id)
+        self._workflow_outputs[workflow_id] = latest_output
         wf_store = workflow_storage.WorkflowStorage(workflow_id, self._store)
         wf_store.save_workflow_meta(
             common.WorkflowMetaData(common.WorkflowStatus.RUNNING))
-        self._step_status[workflow_id] = {}
-        return output
+        self._step_status.setdefault(workflow_id, {})
+        return result.state
```

Comment:
The question here: if it's a normal workflow, will it be wrong? should we return `result.output`? Does this means here `result.state == result.output`?

Reply:
state is for both workflow and virtual actor writer (state is the object to persist)